examples/llm/src/models/mosaic_gpt/mosaic_gpt.py

```diff
@@ -290,11 +334,13 @@ def forward(
             assert isinstance(self.transformer.emb_drop, nn.Module)  # pyright
             x = self.transformer.emb_drop(x_shrunk)
 
-        attn_bias, attention_mask = self._attn_bias(
-            device=x.device,
-            dtype=x.dtype,
-            attention_mask=attention_mask,
-            prefix_mask=prefix_mask)
+        with torch.no_grad():
```

Comment:
can we shove this into the `self._attn_bias` fn or better yet make it a decorator over that fn (see example [here](https://pytorch.org/docs/stable/generated/torch.no_grad.html) with decorator)

ex
```
    @torch.no_grad()
    def _attn_bias(self,
                   device,
                   dtype,
                   attention_mask: Optional[torch.ByteTensor] = None,
                   prefix_mask: Optional[torch.ByteTensor] = None):
                   prefix_mask: Optional[torch.ByteTensor] = None,
                   sequence_id: Optional[torch.LongTensor] = None):
```
when you define the fn above